rust/lance-encoding/src/decoder.rs

```diff
@@ -1182,12 +1182,41 @@ pub trait PhysicalPageDecoder: Send + Sync {
     /// * `rows_to_skip` - how many rows to skip (within the page) before decoding
     /// * `num_rows` - how many rows to decode
     /// * `dest_buffers` - the output buffers to decode into
+    // fn decode_into(
+    //     &self,
+    //     rows_to_skip: u32,
+    //     num_rows: u32,
+    //     dest_buffers: &mut [BytesMut],
+    // ) -> Result<()>;
+
+    /// Combined function to update the capacity required to represent the requested data
+    /// and decode the data into the requested buffers.
+    ///
```

Comment:
No need to describe it as a "combined function".  The fact that we are combining two functions into one will hopefully be lost to history.  The first sentence of a method doc should be a brief summary of what the method does.

```suggestion
    /// Decode data into buffers
    ///
    /// This may be a simple zero-copy from a disk buffer or could involve complex decoding
    /// such as decompressing from some compressed representation.
```